nuheat/thermostat.py

```diff
@@ -174,41 +178,119 @@ def schedule_mode(self, mode):
 
+    @property
+    def hold_time(self):
+        """
+        Return a datetime for the current temporary hold time (or None)
+        """
+        if self._schedule_mode == config.SCHEDULE_TEMPORARY_HOLD:
+            return self._hold_time
+        else:
+            return None
+
+    @hold_time.setter
+    def hold_time(self, hold):
+        """
+        Set a temporary hold time (and change schedule_mode to Temporary Hold).
+
+        :param hold: datetime for temporary hold_time.
+        """
+        hold_gmt = hold.astimezone(timezone(timedelta(0), 'GMT'))
+
+        if hold_gmt < datetime.now(timezone.utc):
+            raise Exception("Invalid hold_time - must be in the future.")
+
+        hold_str = hold_gmt.strftime("%a, %d %b %Y %H:%M:%S %Z")
+        post_data = {
+            "ScheduleMode": config.SCHEDULE_TEMPORARY_HOLD,
+            "HoldSetPointDateTime": hold_str
+        }
+        self.set_data(post_data)
+
+    @property
+    def next_schedule_event(self):
+        """
+        Return dictionary containing information about the next scheduled
+        event.
+        """
+        if self._data is None:
+            return None
+
+        # Get thermostat's timezone from server data
+        tstat_tzs = self._data.get("TZOffset", "")
+        tstat_tz = time.fromisoformat("00:00:00" + tstat_tzs).tzinfo
+
+        # Convert now to thermostat's timezone, so day/next_day will be right
+        now = datetime.now().astimezone(tstat_tz)
+
+        # Find the first active event in the next week that is after now.
+        # Note: We start with yesterday (-1), in case yesterday's "Sleep" time
+        # is set for today.
+        for add_days in range(-1, 8):
+            day = now.date() + timedelta(days=add_days)
+            next_day = day + timedelta(days=1)
+            dayofweek = day.weekday()
+            for event in self._data.get("Schedules")[dayofweek].get("Events"):
+                # Times in thermostat schedule are relative to TZOffset
+                event_time = time.fromisoformat(event.get("Clock") + tstat_tzs)
+                if (event.get("ScheduleType") == 3 and
+                    event_time <= time(3, 0, 0, 0, tstat_tz)):
+                    # Special case: Thermostat schedule will accept times
+                    # between midnight and 3am for the "Sleep" time.  These
+                    # actually occur on the following day, based on empirical
+                    # testing.
+                    event_dt = datetime.combine(next_day, event_time)
+                else:
+                    event_dt = datetime.combine(day, event_time)
+                if event.get("Active") and (now < event_dt):
+                    # Convert back to local time before returning result
+                    event_dt = event_dt.astimezone(datetime.now().tzinfo)
+                    temp_floor = event.get("TempFloor")
+                    return {"Time": event_dt, "NuheatTemperature": temp_floor}
```

Comment:
sorry for the last minute nitpick, but can we change this to just `Temperature`? there are other brands that can use this same API (see PR #5), so i'd like to avoid namespacing things with brands as much as possible.

Reply:
At the moment, the value being returned is a temperature value that would need to be passed to either `nuheat_to_fahrenheit` or `nuheat_to_celsius` to get a real temperature.  Do you think it would still be clear that this conversion is still necessary?

Reply:
ohhh, got it. this makes sense then!